Lombiq.Tests.UI/Extensions/VisualVerificationUITestContextExtensions.cs

```diff
@@ -400,22 +401,29 @@ public static class VisualVerificationUITestContextExtensions
         // We take a screenshot of the element area. This will be compared to a baseline image.
         using var elementImageOriginal = context.TakeElementScreenshot(element).ShouldNotBeNull();
 
-        // Checking the size of captured image.
-        try
-        {
-            elementImageOriginal.Width
-                .ShouldBeGreaterThanOrEqualTo(cropRegion.Left + cropRegion.Width);
-            elementImageOriginal.Height
-                .ShouldBeGreaterThanOrEqualTo(cropRegion.Top + cropRegion.Height);
-        }
-        catch
+        // Checking the dimensions of captured image. This needs to happen before any other comparisons, because that
+        // can only be done on images with the same dimensions.
+        var cropWidth = cropRegion.Left + cropRegion.Width;
+        var cropHeight = cropRegion.Top + cropRegion.Height;
+        if (elementImageOriginal.Width < cropWidth || elementImageOriginal.Height < cropHeight)
         {
+            var cropRegionName = regionOfInterest == null ? "baseline image" : "selected region of interest";
+            var message = $"The dimensions of the captured element ({elementImageOriginal.Width.ToTechnicalString()}" +
+                $"px x {elementImageOriginal.Height.ToTechnicalString()}px) are smaller than the dimensions of the " +
+                $"{cropRegionName} ({cropWidth.ToTechnicalString()}px x {cropHeight.ToTechnicalString()}px). This " +
+                "can happen if due to a change in the app the captured element got smaller than before, or if the " +
+                $"{cropRegionName} is mistakenly too large.";
+
             if (approvedContext != null)
             {
-                context.SaveSuggestedImage(element, approvedContext.BaselineImagePath, approvedContext.BaselineFileName);
+                message += " The suggested baseline image with a screenshot of the captured element was saved to the " +
+                    "failure dump. Compare this with the original image used by the test and if suitable, use it as " +
+                    "the baseline going forward.";
+                context.SaveSuggestedImage(
```

Comment:
If `approvedContext` is `null`, the image not get added to `FailureDump`.
See how the `elementImageOriginal` named in the `FailureDump`:
https://github.com/Lombiq/UI-Testing-Toolbox/blob/b43d44ed235e65a487e2f66053f91e17e00cad9c/Lombiq.Tests.UI/Extensions/VisualVerificationUITestContextExtensions.cs#L476

Also you can add these images anywhere to `FailureDump` since these items are only created on the filesystem if the test fails.

Reply:
You can use `AddImageToFailureDump` here.

Reply:
I didn't really follow what you first meant, but now I think we have some good refactoring.